cmd/collect_signals/github/legacy/issues.go

```diff
@@ -0,0 +1,62 @@
+package legacy
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/go-github/v44/github"
+	"github.com/ossf/criticality_score/internal/githubapi"
+)
+
+type IssueState string
+
+const (
+	IssueStateAll    = "all"
+	IssueStateOpen   = "open"
+	IssueStateClosed = "closed"
+)
+
+// FetchIssueCount returns the total number of issues for a given repo in a
+// given state, across the past lookback duration.
+//
+// This count includes both issues and pull requests.
+func FetchIssueCount(ctx context.Context, c *githubapi.Client, owner, name string, state IssueState, lookback time.Duration) (int, error) {
+	opts := &github.IssueListByRepoOptions{
+		Since:       time.Now().UTC().Add(-lookback),
+		State:       string(state),
+		ListOptions: github.ListOptions{PerPage: 1}, // 1 result per page means LastPage is total number of records.
+	}
+	is, resp, err := c.Rest().Issues.ListByRepo(ctx, owner, name, opts)
+	if err != nil {
+		return 0, err
+	}
+	if resp.NextPage == 0 {
+		return len(is), nil
+	}
+	return resp.LastPage, nil
+}
+
+// FetchIssueCommentCount returns the total number of comments for a given repo
+// across all issues and pull requests, for the past lookback duration.
+//
+// If the exact number if unable to be returned because there are too many
+// results, a TooManyResultsError will be returned.
+func FetchIssueCommentCount(ctx context.Context, c *githubapi.Client, owner, name string, lookback time.Duration) (int, error) {
+	since := time.Now().UTC().Add(-lookback)
+	opts := &github.IssueListCommentsOptions{
+		Since:       &since,
+		ListOptions: github.ListOptions{PerPage: 1}, // 1 result per page means LastPage is total number of records.
+	}
+	cs, resp, err := c.Rest().Issues.ListComments(ctx, owner, name, 0, opts)
+	// The API returns 5xx responses if there are too many comments.
+	if c := githubapi.ErrorResponseStatusCode(err); 500 <= c && c < 600 {
+		return 0, TooManyResultsError
```

Comment:
This seems inconsistent with the other implementations' behaviour (e.g. contributor count), which seems to hide this detail from the caller and just return the maximum number. 

i.e. 

```go
	cs, resp, err := c.Rest().Repositories.ListContributors(ctx, owner, name, opts)
	if errorTooManyContributors(err) {
		return MaxContributorLimit, nil
	}
```

Would that make more sense here too?

Reply:
I'm not a huge fan of this difference, but it currently exists this way due to the usage of the results from `FetchIssueCount()` to generate the frequency (i.e. average comments per issue).

I have an refactoring in mind to improve the API around a struct{} that could also be used to reduce the number of args to the methods and provide some caching - making it possible to move this into a more logical location.

For the time being I'd be happy to get this merged and do some future refactoring soon (I've added a TODO).

Reply:
ack.